requirements-test.txt

```diff
@@ -6,3 +6,8 @@ scikit-learn==1.2.2 ; python_version == '3.8'
 scikit-learn==1.3.1 ; python_version >= '3.9'
 pandas==2.0.1 ; python_version == '3.8'
 pandas==2.1.1 ; python_version >= '3.9'
+xgboost==1.7.6; python_version <= '3.9'
```

Comment:
should it be test vs requirements-test-optional.txt?

Reply:
Needed here, in `test_model_builders.py` (part of Intel CI) we are now comparing against XGBoost, LightGBM, CatBoost reference predictions.

Reply:
well, same as it was for test_xgboost_mb

The problem is that  test requirements imply that this is test requirements for sklearnex while they are not.  

tests-optional is not great name and might be worth renaming to target daal4py only.  This might be also good for testing steps separation so daal4py and sklearnex would be tested in separate steps.  
This is also true for tests placement - now current design is that we have all test files in same place and xgboost_mb check if xgboost package is available and skip test if it's not. 

The better approach would be in separation of test files between scikit-learn and daal4py testing and making them non optional

Reply:
More on that
Here are daal4py test deps  and test commands
https://github.com/conda-forge/daal4py-feedstock/blob/main/recipe/meta.yaml#L42

and this is scikit
https://github.com/conda-forge/scikit-learn-intelex-feedstock/blob/main/recipe/meta.yaml#L31

those have to be cleaned for 2024.0 in terms of commands but from tests perspective we don't need to bring GBT dependencies to scikit

Reply:
let's keep single requirements-test file and drop one specific for dppy and test-optional?

Reply:
I actually can't find a place where `requirements-dppy.txt` is used. I'm fine with removing it.

Reply:
Checked in with @samir-nasibli he wants to keep it for now. We can always clean up separately. Does not have to happen in this PR.